pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -645,17 +649,66 @@ func (a *ApplicationRestoreController) downloadCRD(
 	if err = json.Unmarshal(crdData, &crds); err != nil {
 		return err
 	}
+	if err = json.Unmarshal(crdData, &crdsV1); err != nil {
+		return err
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("error getting cluster config: %v", err)
+	}
+
+	client, err := apiextensionsclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	regCrd := make(map[string]bool)
 	for _, crd := range crds {
 		crd.ResourceVersion = ""
-		if err := apiextensions.Instance().RegisterCRD(crd); err != nil && !errors.IsAlreadyExists(err) {
-			return err
+		if _, err := client.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd); err != nil && !errors.IsAlreadyExists(err) {
+			regCrd[crd.GetName()] = false
+			logrus.Warnf("error registering crds %v,%v", crd.GetName(), err)
+			continue
 		}
-		if err := apiextensions.Instance().ValidateCRD(apiextensions.CustomResource{
-			Plural: crd.Spec.Names.Plural,
-			Group:  crd.Spec.Group}, validateCRDTimeout, validateCRDInterval); err != nil {
-			return err
+		// wait for crd to be ready
+		if err := k8sutils.ValidateCRD(client, crd.GetName()); err != nil {
+			logrus.Warnf("Unable to validate crds %v,%v", crd.GetName(), err)
 		}
+
 	}
+
+	for _, crd := range crdsV1 {
+		if val, ok := regCrd[crd.GetName()]; ok && val {
```

Comment:
I don't see the key for this map being set to true for any key. I only see it being set to false on line 669

Reply:
thanks for catching this, added at line no 668